### PR TITLE
Allow cURL Statistics option in write-http plugin

### PR DESCRIPTION
**Pull request by n2yen**
write-http can optionally configure cURL Statistics as per other cURL based plugins

ChangeLog: write_http plugin: add ability to configure cURL Statistics

### CODE REVIEW

**Comment by n2yen** [comment]
ChangeLog: write_http plugin: add ability to configure cURL Statistics

**Comment by mrunge** [comment]
Thank you for this proposal, I'll get back to this asap.

**Comment by n2yen** [comment]
thanks @mrunge sure !

File: src/collectd.conf.pod [hunk 1]
```diff
@@ -10138,6 +10138,12 @@ and the size of B<BufferSize>. The optimal value to set B<Timeout> to is
 slightly below this interval, which you can estimate by monitoring the network
 traffic between collectd and the HTTP server.
 
+=item B<E<lt>StatisticsE<gt>>
```
Comment (kwiatrox) [inline src/collectd.conf.pod]: I think it is better to add it after LogHttpError item, and before the note about Timeout.

File: src/write_http.c [hunk 2]
```diff
@@ -130,6 +132,9 @@ static int wh_post_nolock(wh_callback_t *cb, char const *data) /* {{{ */
 
   wh_log_http_error(cb);
 
+  if (cb->curl_stats != NULL)
+    curl_stats_dispatch(cb->curl_stats, cb->curl, NULL, "write_http", cb->name);
```
Comment (kwiatrox) [inline src/write_http.c]: Check return code from curl_stats_dispatch.

File: src/write_http.c [hunk 3]
```diff
@@ -317,6 +322,8 @@ static void wh_callback_free(void *data) /* {{{ */
     cb->curl = NULL;
   }
 
+  curl_stats_destroy(cb->curl_stats);
```
Comment (kwiatrox) [inline src/write_http.c]: Add:
cb->curl_stats = NULL;

**Comment by n2yen** [comment]
thanks for your feedback @kwiatrox - I've updated based on your comments.
cheers.

**Comment by n2yen** [comment]
thanks for the review @kwiatrox !

**Comment by n2yen** [comment]
Hi @kwiatrox , just checking in, is there anything else we need for review approval?

**Comment by n2yen** [comment]
thanks @dago !

**Pull request merged**